appengine/src/services/sync.js

```diff
@@ -97,47 +101,63 @@ module.exports = function (BLOCKCHAIN_SERVER) {
     }
   }
   const getValidatorInfoData = async address => {
-    const res = await apiClient.post(
-      '/',
-      bodyParams('hmy_getValidatorInformation', address)
-    )
+    try {
+      const res = await apiClient.post(
+        '/',
+        bodyParams('hmy_getValidatorInformation', address)
+      )
 
-    if (isNotEmpty(res.data.result)) {
-      let selfStake = 0
-      let totalStake = 0
-      if (cache[DELEGATIONS_BY_VALIDATOR][address]) {
-        const elem = cache[DELEGATIONS_BY_VALIDATOR][address].find(
-          e => e.validator_address === e.delegator_address
-        )
-        if (elem) {
-          selfStake = elem.amount
+      if (isNotEmpty(res.data.result)) {
+        let selfStake = 0
+        let totalStake = 0
+        if (cache[DELEGATIONS_BY_VALIDATOR][address]) {
+          const elem = cache[DELEGATIONS_BY_VALIDATOR][address].find(
+            e => e.validator_address === e.delegator_address
+          )
+          if (elem) {
+            selfStake = elem.amount
+          }
+          totalStake = cache[DELEGATIONS_BY_VALIDATOR][address].reduce(
+            (acc, val) => acc + val.amount,
+            0
+          )
         }
-        totalStake = cache[DELEGATIONS_BY_VALIDATOR][address].reduce(
-          (acc, val) => acc + val.amount,
-          0
-        )
-      }
 
-      // fields below are included in the validator.
-      // * signed_blocks
-      // * blocks_should_sign
-      // * total_one_staked
-
-      const validatorInfo = {
-        active: !!cache[ACTIVE_VALIDATORS].includes(address),
-        self_stake: selfStake,
-        total_stake: totalStake,
-        // TODO(minh) fix it.
-        signed_blocks: 50,
-        blocks_should_sign: 100,
-        total_one_staked: 4,
-        ...res.data.result
-      }
+        // fields below are included in the validator.
+        // * signed_blocks
+        // * blocks_should_sign
+        // * total_one_staked
+        const utcDate = new Date(Date.now())
+
+        const validatorInfo = {
+          active: !!cache[ACTIVE_VALIDATORS].includes(address),
+          self_stake: selfStake,
+          total_stake: totalStake,
+          // TODO(minh) fix it.
+          signed_blocks: 50,
+          blocks_should_sign: 100,
+          total_one_staked: 4,
+          uctDate: utcDate,
+          ...res.data.result
+        }
+
+        cache[VALIDATOR_INFO][address] = validatorInfo
 
-      cache[VALIDATOR_INFO][address] = validatorInfo
+        // Calculating cache[VALIDATOR_INFO_HISTORY]
+        const timeIndex = Math.floor(Math.floor(utcDate.getTime() / 1000)/60)
+        if (!cache[VALIDATOR_INFO_HISTORY][address]) {
+          cache[VALIDATOR_INFO_HISTORY][address] = new Map();
+        }
+        cache[VALIDATOR_INFO_HISTORY][address][timeIndex] = validatorInfo
```

Comment:
I think will better save this data in database

Also - we need to save data by day